jobs/buildhub/inventory_to_records.py

```diff
@@ -107,9 +107,13 @@ class JSONFileNotFound(Exception):
         raise
 
 
-async def fetch_listing(session, url):
+async def fetch_listing(session, url, retry_on_notfound=False):
     try:
-        data = await fetch_json(session, url)
+        data = await fetch_json(
+            session,
+            url,
+            retry_on_notfound=retry_on_notfound,
+        )
         return data['prefixes'], data['files']
     except (aiohttp.ClientError, KeyError, ValueError) as e:
         raise ValueError("Could not fetch '{}': {}".format(url, e))
```

Comment:
We may want to add a test here https://github.com/peterbe/buildhub/blob/6ac3fa0f1708558f55faeb5a205fd66bf975dbf4/jobs/tests/test_inventory_to_records.py#L192 ;)

Reply:
I'm actually not a fan of this. It's not great that it lumps together all of these different exceptions. 

HTTP problems should probably be separated from problems with the content. The HTTP problems might sometimes be something you want to retry but if the content is busted it will never help to retry. 

Let we discussed. Hopefully this PR will at least solve one of the tracebacks we saw in Sentry.